src/Stats.h

```diff
@@ -17,17 +18,27 @@ class Stats
 public:
     Stats();
     void reset(StatsEnum stat);
-    void increment(StatsEnum stat, int increase = 1, float floatIncrease = 1.0);
-    void incrementFloat(StatsEnum stat, float increase = 1.0);
+    void resetAll();
+    void addDistance(int distance = 1);
+    void addTurn();
     void bindText(StatsEnum stat, QLineEdit* uiText);
-    void incrementEffectiveDistance(int distance);
-
+    void startRun();
+    void finishRun(bool reachedGoal);
```

Comment:
Pro tip: avoid boolean arguments, because it's hard to know what they do when reading the calling code. Instead, just create two methods: one for indicating that the goal was reached, and one for indicating that the starting cell was re-entered.